### PR TITLE
[FLINK-33236][config] Deprecate the unused high-availability.zookeeper.path.running-registry option

### PR DESCRIPTION
**Pull request by X-czh**
## What is the purpose of the change

The running registry subcomponent of Flink HA has been removed in [FLINK-25430](https://issues.apache.org/jira/browse/FLINK-25430) and the "high-availability.zookeeper.path.running-registry" option is of no use after that. We should Deprecate the option and regenerate the config doc to remove the relevant descriptions to avoid user's confusion.

## Brief change log

Deprecate the unused high-availability.zookeeper.pth.running-registry option and remove the relevant doc.

## Verifying this change

This change is a trivial code cleanup.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java
Comment (XComp) [inline flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:170]: Isn't the right process to deprecate the config option and remove it as part of 2.0? ...since configs are considered public API. :thinking:

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks for picking that up. That one seem to have been missed. I'm just wondering: We might want to deprecate the option rather than deleting it right away. Config options are considered public API. Therefore, the removal should happen in 2.0

File: flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java
Comment (X-czh) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:170]: I agree that this Isn't the right process to deprecate the config option. What's special about it is that usually when we deprecate a config, the config can still take effect, but in this case, the config has no effect. So we might better just delete it to avoid user confusion, WDYT?

Comment (X-czh) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:170]: I just found a similar case: the option high-availability.job.delay is deprecated and the comment states clearly that it has no effect on Flink any longer. That seems like the right process to do it. I'll follow the approach and update the PR. Thank you for pointing that out!

Comment (XComp) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:170]: Yeah, I see your point. The worst thing that could happen as far as I can think is that the field is used in some downstream project. I guess that would be a reason why we would follow the proper deprecation process. The only documentation I found about that is what's stated in the kind of old [Best Practices and Lessons Learned](https://cwiki.apache.org/confluence/display/FLINK/Best+Practices+and+Lessons+Learned)

File: flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java
Comment (XComp) [inline flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: Maybe, add a description stating that the config parameter is deprecated and not used anymore.

File: docs/layouts/shortcodes/generated/expert_high_availability_zk_section.html
Comment (XComp) [inline docs/layouts/shortcodes/generated/expert_high_availability_zk_section.html:59]: The documentation needs to be regenerated

**Review by XComp — CHANGES_REQUESTED** [review]

File: docs/layouts/shortcodes/generated/expert_high_availability_zk_section.html
Comment (X-czh) [reply docs/layouts/shortcodes/generated/expert_high_availability_zk_section.html:59]: I've already regenerated the doc

File: flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java
Comment (X-czh) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: The description on deprecation has already been added in the comment, do you mean to add description using the . withDescription as well?

Comment (XComp) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: Yes, so that users who check the website would be also aware of the change.

Comment (X-czh) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: But the doc generator won't generate docs for deprecated options, so users won't see this on the website.

Comment (X-czh) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: I think we'd better generate docs for deprecated options as well to offer proper upgrading guidance in the doc, otherwise users have to either carefully read the release notes or to directly look up the source code. I've created a JIRA to track this: https://issues.apache.org/jira/browse/FLINK-33240. Maybe we can work on that issue first, and come back to this one.

Comment (XComp) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: hm, I wanted to reproduce it but failed. How did you generate the docs?
I tried the what's documented in [docs/README.md:62](https://github.com/apache/flink/blob/5c1e9f3b1449cb77276d578b344d9a69c7cf9a3c/docs/README.md#L62):
```bash
mvn -Pgenerate-config-docs install -Dfast -DskipTests
```
and what's documented in [flink-docs/README.md:44](https://github.com/apache/flink/blob/7bebd2d9fac517c28afc24c0c034d77cfe2b43a6/flink-docs/README.md#L44)
```bash
mvn package -Dgenerate-config-docs -pl flink-docs -am -nsu -DskipTests
```
Both approaches didn't regenerated the HTML files. What am I missing? :thinking:

Comment (XComp) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: I was wrong. I didn't properly revert the files before running the command again.

File: docs/layouts/shortcodes/generated/expert_high_availability_zk_section.html
Comment (XComp) [reply docs/layouts/shortcodes/generated/expert_high_availability_zk_section.html:59]: Yeah, I wasn't aware of the feature where we remove the documentation for deprecated options.

File: flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java
Comment (X-czh) [reply flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java:173]: I've added description in a new commit

**Review by XComp — APPROVED** [review]
LGTM :+1: Let's wait for CI before merging the change. Thanks again, @X-czh

**Pull request merged**
